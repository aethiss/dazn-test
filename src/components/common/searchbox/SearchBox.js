import React from 'react'
import style from './SearchBox.css'

const SearchBox = (props) => {
  return (
      <input
        type="search"
        placeholder="Search"
        onKeyDownCapture={(ele) => props.updateSearch(ele)}
      />
  )
}

export default SearchBox
