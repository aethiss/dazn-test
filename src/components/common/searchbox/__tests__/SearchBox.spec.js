import React from 'react'
import { shallow, mount } from "enzyme"

import SearchBox from '../SearchBox'

const mockProps = {
  updateSearch: jest.fn()
}

describe('@SearchBox', () => {
  describe('renders', () => {
    it('should render a input form', () => {
      const component = shallow(<SearchBox {...mockProps} />)
      expect(component).toMatchSnapshot()
      expect(component.find('input').props().type).toBe('search')
    })
  })
  describe('actions', () => {
    it('should call updateSearch after enter', () => {
      const component = mount(<SearchBox {...mockProps} />)
      const input = component.find('input')
      input.simulate('change', { target: { value: 'abcdefg'} })
      input.simulate('keydown', { keyCode: 13 })
      expect(component.props().updateSearch).toHaveBeenCalled()
    })
  })
})
