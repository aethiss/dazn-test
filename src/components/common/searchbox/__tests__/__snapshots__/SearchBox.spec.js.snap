// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@SearchBox renders should render a input form 1`] = `
<input
  onKeyDownCapture={[Function]}
  placeholder="Search"
  type="search"
/>
`;
